include/pmacc/math/complex/Bessel.tpp

```diff
@@ -0,0 +1,276 @@
+                    cq0 += b[ k ] * pmMath::pow(
+                        z1,
+                        float_T( -2.0 * k - 3.0 )
+                    );
+                }
+                complex_T const cu = pmMath::sqrt( float_T( M_2_PI ) / z1 );
+                cj0 = cu * ( cp0 * pmMath::cos( ct1 ) - cq0 * pmMath::sin( ct1 ) );
+            }
+            return cj0;
+        }
+    };
+
+    template<
+        typename T_Type,
+        typename T_TableA,
+        typename T_TableB,
+        typename T_TableA1,
+        typename T_TableB1
+    >
+    struct Cbesselj1Base
+    {
+        using result = pmacc::math::Complex< T_Type >;
+        using complex_T = pmacc::math::Complex< T_Type >;
+        using float_T = T_Type;
+        float_T const eps = float_T( 1.0e-15 );
+        float_T const el = float_T( 0.5772156649015329 );
+
+        complex_T const cii = complex_T( 0, 1 );
+        complex_T const cone = complex_T( 1, 0 );
+        complex_T const czero = complex_T( 0, 0 );
+
+        HDINLINE result operator( )( complex_T const & z )
+        {
+            T_TableA a;
+            T_TableB b;
+            T_TableA1 a1;
+            T_TableB1 b1;
+            result cj1;
+
+            float_T const a0 = pmMath::abs( z );
+            float_T const z2 = z * z;
+            complex_T z1 = z;
+            if ( a0 == float_T( 0.0 ) )
+            {
+                cj1 = czero;
+                return cj1;
+            }
+            if ( z.get_real() < float_T( 0.0 ) ) z1 = -z;
+            if ( a0 <= float_T( 12.0 ) )
+            {
+                cj1 = cone;
+                complex_T cr = cone;
+                for ( uint32_t k = 1u; k <= 40u; k++ )
+                {
+                    cr *= float_T( -0.25 ) * z2 / ( k * ( k + float_T( 1.0 ) ) );
+                    cj1 += cr;
+                    if ( pmMath::abs( cr ) < pmMath::abs( cj1 ) * eps) break;
+                }
+                cj1 *= float_T( 0.5 ) * z1;
+            }
+            else {
+                uint32_t kz;
+                if ( a0 >= float_T( 50.0 ) ) kz = 8u;      // can be changed to 10
+                else if (a0 >= float_T( 35.0 ) ) kz = 10u; //   "      "     "  12
+                else kz = 12u;                             //   "      "     "  14
+                complex_T const cu = pmMath::sqrt( float_T( M_2_PI ) / z1 );
+                complex_T const ct2 = z1 - float_T( 0.75 ) * float_T( M_PI );
+                complex_T cp1 = cone;
+                for (uint32_t k = 0u; k < kz; k++ )
+                {
+                    cp1 += a1[ k ] * pmMath::pow(
+                        z1,
+                        float_T( -2.0 * k - 2.0 )
+                    );
+                }
+                complex_T cq1 = float_T( 0.375 ) / z1;
+                for (uint32_t k = 0u; k < kz; k++ )
+                {
+                    cq1 += b1[ k ] * pmMath::pow(
+                        z1,
+                        float_T( -2.0 * k - 3.0)
```

Comment:
please write `float_T( -2.0 ) * float_T( k ) - float_T( 3.0 )` to avoid calculations with the wrong precision